### ISSUE TITLE
aws-sdk calls don't work

### ISSUE THREAD

**Issue by ammulder** [stevejhiggs/aws-lambda-typescript#3]
I have a JavaScript lambda where I make a single Cognito call to check whether the aws-sdk is working:

```
let Cognito = require('aws-sdk/clients/cognitoidentityserviceprovider');

exports.handler = (event, context, callback) => {
    let client = new Cognito({ apiVersion: '2016-04-19', region: 'us-east-1' });

    let params = {
        AuthFlow: 'REFRESH_TOKEN_AUTH',
        ClientId: 'MY_CLIENT_ID',
        UserPoolId: 'us-east-1_MY_POOL_ID',
        AuthParameters: {
            REFRESH_TOKEN: 'some bogus value'
        }
    };
    client.adminInitiateAuth(params, function(err, result) {
        console.log(err);
        console.log(result);
        // Test complete.  Now go on and do the regular processing
    });
```
I expect to get an error there along the lines of "invalid token" because I'm not using a real refresh token.  I just want to log the output of the call and then go on to do the regular processing for the Lambda.  Here's what I get in the log (as expected):
```
NotAuthorizedException: Invalid Refresh Token
at Request.extractError (/var/task/node_modules/aws-sdk/lib/protocol/json.js:48:27)
```

Here's my TypeScript version:
```
import {Callback, Handler} from 'aws-lambda';
import * as CognitoIdentityServiceProvider from "aws-sdk/clients/cognitoidentityserviceprovider";

const func: Handler = (event, context, callback: Callback) => {
  let client = new CognitoIdentityServiceProvider({ apiVersion: '2016-04-19', region: 'us-east-1' });

  let params = {
    AuthFlow: 'REFRESH_TOKEN_AUTH',
    ClientId: 'MY_CLIENT_ID',
    UserPoolId: 'us-east-1_MY_POOL_ID',
    AuthParameters: {
        REFRESH_TOKEN: 'some bogus value'
    }
  };
  client.adminInitiateAuth(params, function(err, result) {
    console.log(err);
    console.log(result);
    // Test complete.  Now go on and do the regular processing
  });
```
I expected the same result.  However, when I make the same call in the TypeScript version of the Lambda, I get this:
```
{ CredentialsError: Missing credentials in config
    at r (/var/task/index.js:1:143946)
    at constructor.getCredentials (/var/task/index.js:1:144474)
    ...
```
In other words, it fails to recognize that the current user/identity/whatever can even call Cognito.  The two Lambdas use the same execution role -- so I'm pretty sure it's not actually a permissions problem, but that somehow the aws-sdk isn't working properly.

I saw [one documentation page](https://aws.amazon.com/blogs/developer/using-webpack-and-the-aws-sdk-for-javascript-to-create-and-bundle-an-application-part-1/) saying you must set up a JSON loader in your Webpack config for the aws-sdk to work (and [another saying this is now automatic](https://github.com/webpack-contrib/json-loader)), and another saying you had to somehow exclude the aws-sdk in your configuration, but these were somewhat out of date.  I'm not yet sure what the real issue is.

**Comment by ammulder** [comment]
Hmm, I see [an answer here](https://stackoverflow.com/questions/32253362/how-do-i-build-a-single-js-file-for-aws-lambda-nodejs-runtime) saying that Lambda already has the aws-sdk in its environment.  But Webpack doesn't seem to exclude it just because it's in my devDependencies not my dependencies.  Is there a way to make Webpack access aws-sdk from the existing environment instead of bundling its own copy?

**Comment by ammulder** [comment]
OK, this one seems to be resolved by adding this to the webpack.config.js in gulp-webpack-typescript-pipeline:
```
      externals: [/aws-sdk/],
```

Maybe you can add a setting for externals to gulp-webpack-typescript-pipeline and then set it to /aws-sdk/ by default in aws-lambda-typescript?

**Comment by stevejhiggs** [comment]
sounds good to me

**Comment by stevejhiggs** [comment]
would you mind trying out v0.0.13? That should fix your issue
